sources/api/pluto/src/api.rs

```diff
@@ -1,76 +1,183 @@
-        })
-    }
+#[derive(Deserialize)]
+struct SettingsView {
+    pub os: Os,
+    pub settings: View,
+}
+
+#[derive(Debug, Snafu)]
+pub(crate) enum Error {
+    #[snafu(display("Failed to call apiclient: {}", source))]
+    CommandFailure { source: std::io::Error },
+    #[snafu(display("apiclient execution failed: {}", reason))]
+    ExecutionFailure { reason: String },
+    #[snafu(display("Deserialization of configuration file failed: {}", source))]
+    Deserialize {
+        #[snafu(source(from(serde_json::Error, Box::new)))]
+        source: Box<serde_json::Error>,
+    },
 }
 
-/// This dummy code is compiled when the `sources` workspace is being compiled for non `aws-k8s-*`
-/// variants.
-#[cfg(not(variant_family = "aws-k8s"))]
-mod inner {
-    use super::*;
-    use snafu::Snafu;
+pub(crate) async fn client_command<I, S>(args: I) -> Result<Vec<u8>>
+where
+    I: IntoIterator<Item = S>,
+    S: AsRef<OsStr>,
+{
+    let result = Command::new("/usr/bin/apiclient")
+        .args(args)
+        .output()
+        .await
+        .context(CommandFailureSnafu)?;
 
-    #[derive(Debug, Snafu)]
-    pub(crate) enum Error {
-        #[snafu(display(
-            "The get_aws_k8s_info function is only compatible with aws-k8s variants"
-        ))]
-        WrongVariant,
-    }
+    ensure!(
+        result.status.success(),
+        ExecutionFailureSnafu {
+            reason: String::from_utf8_lossy(&result.stderr)
+        }
+    );
 
-    pub(crate) async fn get_aws_k8s_info() -> Result<AwsK8sInfo> {
-        WrongVariantSnafu.fail()
+    Ok(result.stdout)
+}
+
+/// Gets the info that we need to know about the EKS cluster from the Bottlerocket API.
+pub(crate) async fn get_aws_k8s_info() -> Result<AwsK8sInfo> {
+    let view_str = client_command(&[
+        "get",
+        "os.variant_id",
+        "settings.aws.region",
+        "settings.network.http-proxy",
+        "settings.network.no-proxy",
+        "settings.kubernetes.cluster-name",
+        "settings.kubernetes.cluster-dns-ip",
+        "settings.kubernetes.node-ip",
+        "settings.kubernetes.max-pods",
+        "settings.kubernetes.provider-id",
+        "settings.kubernetes.hostname-override",
+    ])
+    .await?;
+    let view: SettingsView =
+        serde_json::from_slice(view_str.as_slice()).context(DeserializeSnafu)?;
+    // Before returning the data, let's ensure to sync the network
+    // proxy settings to the environment
+    if let Some(https_proxy) = view
+        .settings
+        .network
+        .as_ref()
+        .and_then(|x| x.https_proxy.as_ref())
+    {
+        env::set_var("https_proxy", https_proxy.clone());
+        env::set_var("HTTPS_PROXY", https_proxy.clone());
     }
+    if let Some(no_proxy) = view
+        .settings
+        .network
+        .as_ref()
+        .and_then(|x| x.no_proxy.as_ref())
+    {
+        env::set_var("no_proxy", no_proxy.clone());
+        env::set_var("NO_PROXY", no_proxy.clone());
```

Comment:
`env::set_var` is unsafe in a multi-threaded context - which we may be here in because we're running in an async runtime.

It's probably OK here because we know that no other tasks are running, but we could avoid the problem by keeping these settings in the struct and passing them to the ec2 / eks functions rather than reading from the environment.